src/averageAges.js

```diff
@@ -55,7 +63,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => onlyWithSon
+    ? person.mother && person.sex === 'm'
+    : person.mother);
+
+  const mothers = people.filter(person =>
+    people.some(child => child.mother === person.name));
+
+  const childMother = children.filter(child =>
+    mothers.some(mother => child.mother === mother.name));
+
+  const averageAge = childMother.reduce(
+    (sum, curr) => sum + (curr.born - people.find(mother =>
+      curr.mother === mother.name).born), 0) / childMother.length;
+
+  return averageAge;
```

Comment:
Мені важкувато, але краще відразу в таких випадках return робити.
Хоча з додатковими змінними мені легше читається
```suggestion
 return childMother.reduce(
    (sum, curr) => sum + (curr.born - people.find(mother =>
      curr.mother === mother.name).born), 0) / childMother.length;
```